fibsem/milling/patterning/plotting.py

```diff
@@ -108,29 +113,188 @@ def _draw_rectangle_pattern(
             p, pixel_size, image_shape
         )
 
-        rect = mpatches.Rectangle(
-            (px - width / 2, py - height / 2),  # bottom left corner
-            width=width,
-            height=height,
-            angle=math.degrees(p.rotation),
-            rotation_point=PROPERTIES["rotation_point"],
-            linewidth=PROPERTIES["line_width"],
-            edgecolor=colour,
-            facecolor=colour,
-            alpha=PROPERTIES["opacity"],
+        patch_collection = PatchCollection(
```

Comment:
The legend doesn't work with this code, if we go down the bitmap -> image mask route can you please change this back to patches.